src/app/pages/ArticlePage/experimentTopStories/helpers.tsx

```diff
@@ -0,0 +1,76 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/react';
+import { OptimoBlock } from '#app/models/types/optimo';
+import { TopStoryItem } from '#app/pages/ArticlePage/PagePromoSections/TopStoriesSection/types';
+import TopStoriesSection from '../PagePromoSections/TopStoriesSection';
+import styles from '../ArticlePage.styles';
+
+export const experimentTopStoriesConfig = {
+  topStoriesExperiment: {
+    variants: {
+      control: 50,
+      show_at_halfway: 50,
+    },
+  },
+};
+
+export const enableExperimentTopStories = ({
+  isAmp,
+  service,
+  pathname,
+}: {
+  isAmp: boolean;
+  service: string;
+  pathname: string;
+}) => {
+  if (!isAmp || !service || !pathname) return false;
+  const urn = pathname.split('/').pop()?.slice(0, -4); // .slice() to remove '.amp' at the end of pathname
```

Comment:
This urn looks like the `id` that we have in the RequestContext. Would it be better to get the id from the [ContextProvider here](pathname) and use that instead of pathname?

Reply:
Great suggestion, much better thanks!